### PR TITLE
chore(azure devops): rename azure credential types

### PR DESCRIPTION
**Pull request by sanderaernouts**
## Hey, I just made a Pull Request!

Renamed `ClientSecret` to `AzureClientSecretCredential` and renamed `ManagedIdentity` to `AzureManagedIdentityCredential`. This makes it more explicit that these are Azure-based credentials.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] ~Added or updated documentation~
- [x] ~Tests for new functionality and regression tests for bug fixes~
- [x] ~Screenshots attached (for UI changes)~
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by sanderaernouts** [comment]
@Rugvip I have split out the rename into a separate PR as you requested in #18213

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice, thank you! 👍 

Just needs an API report update: `yarn build:api-reports packages/integration`

File: .changeset/silly-jeans-wonder.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/integration': patch
+---
+
+Renamed ClientSecret to AzureClientSecretCredential and ManagedIdentity to AzureManagedIdentityCredential
```
Comment (awanlin) [inline .changeset/silly-jeans-wonder.md:5]: This should probably be `minor` and then the practice is to prefix this with `**BREAKING**:`

Here's an example for reference: https://github.com/backstage/backstage/blob/6eb90068a88272601bdc030604bf436b21ae31f5/.changeset/silver-cars-type.md?plain=1#LL7C1-L7C14

**Review by awanlin — CHANGES_REQUESTED** [review]
This looks good to me @sanderaernouts, just a comment about the changeset

File: .changeset/silly-jeans-wonder.md (hunk 1 above)
Comment (Rugvip) [reply .changeset/silly-jeans-wonder.md:5]: No worries, this isn't breaking as these were all added in #17780, which hasn't been released yet. It's why I was eager to ship this before next Tuesday and asked for the separate PR. Thank you for keeping an eye out for this though! 🙏

**Comment by sanderaernouts** [comment]
@Rugvip updated the API report 👍

**Comment by sanderaernouts** [comment]
@Rugvip, this PR needs another look if it is to be merged before tomorrow's release.

**Pull request merged**